workflow/scripts/osemosys_global/transmission/data.py

```diff
@@ -1,51 +1,180 @@
 """Functions to extract and format relevent data for tranmission."""
+import pandas as pd
+import geopy.distance
+
+from sets import get_unique_technologies
 
 def get_years(start: int, end: int) -> range:
     return range(start, end + 1)
 
-def format_transmission_name(df):
-    '''Formats PLEXOS transmission names into OSeMOSYS Global names.
-
-    Args:
-        :param df: Pandas DataFrame with columns 'From' and 'To' describing the 
-               transmission from and to contries. ie. 
+def format_gtd_existing(df):
+    cols = {'pathway' : 'TECHNOLOGY',
+            'from_region' : 'From',
+            'to_region' : 'To', 
+            'VALUE' : 'VALUE'
+            }
+    
+    # Sets the capacity of the line to the average of max and max counter flows.
+    df['VALUE'] = (df['max_flow'] + df['max_counter_flow']) / 2
+    
+    df = df[list(cols.keys())]
+    
+    df = df.rename(columns = cols)
     
-    Returns: 
-        :param df: Same as df_in, except the 'From' and 'To' columns are replaced 
-            with a single 'TECHNOLOGY' column holding OSeMOSYS Global 
-            naming conventions 
+    return df
 
-    Example:
-        df = pd.DataFrame(
-            [[AF-COD, AF-COG, 0.001],
-            [EU-AUT, EU-SVK, 0.004],
-            [AS-LBN, AS-SYR, 0.006]], 
-            columns = ['From', 'To', 'Losses']
-        )
-        pd.DataFrame(
-            [[0.001,TRNCODXXCOGXX],
-            [0.004,TRNAUTXXSVKXX],
-            [0.006,TRNLBNXXSYRXX]] 
-            columns = ['Losses','TECHNOLOGY'])'''
+def format_gtd_planned(df):
+    cols = {'pathway' : 'TECHNOLOGY',
+            'from_region' : 'From',
+            'to_region' : 'To', 
+            'year_planned' : 'YEAR',
+            'VALUE' : 'VALUE'
+            }
+    
+    # Sets the capacity of the line to the average of max and max counter flows.
+    df['VALUE'] = (df['max_flow'] + df['max_counter_flow']) / 2
+    
+    df = df[list(cols.keys())]
+    
+    df = df.rename(columns = cols)
+    
+    return df
 
-    # If from column has length 6 then it's the last three chars plus XX
-    df.loc[df["From"].str.len() == 6, "From"] = (df["From"].str[3:6] + "XX")
+def correct_gtd_data(df_exist, df_plan, region_mapping_dict, 
+                     CUSTOM_TRN_BA_DICT_FROM, CUSTOM_TRN_BA_DICT_TO,
+                     CUSTOM_TRN_BA_MISSING):
```

Comment:
Not a required change; just a FYI. 

According to PEP8, [constants should be defined as all caps](https://peps.python.org/pep-0008/#constants) (as you did for the last three arguments in the constants file), but [function arguments and variables should still follow snakecase formatting](https://peps.python.org/pep-0008/#function-and-variable-names).

Reply:
Fixed for both powerplant and transmission scripts